lib/saluki-components/src/sources/dogstatsd/mod.rs

```diff
@@ -550,60 +615,52 @@ async fn drive_stream(mut stream: Stream, source_context: SourceContext, handler
             }
         }
         metrics.events_received().increment(event_buffer.len() as u64);
-        forward_events(event_buffer, &source_context, &peer_addr, &listen_addr).await;
+        forward_events(
+            event_buffer,
+            maybe_eventd_event_buffer,
+            maybe_service_checks_event_buffer,
+            &source_context,
+            &peer_addr,
+            &listen_addr,
+        )
+        .await;
     }
 }
 
-fn handle_frame(
-    frame: &[u8], codec: &mut DogstatsdCodec<AgentLikeTagMetadataInterceptor>, event_buffer: &mut EventBuffer,
-    origin_detection: bool, peer_addr: &ConnectionAddress,
-) -> Result<(), ParseError> {
-    codec.decode_packet(frame, event_buffer)?;
-
-    // We do one optional enrichment step here, which is to add the client's socket credentials as a tag
-    // on each metric, if they came over UDS. This would then be utilized downstream in the pipeline by
-    // origin enrichment, if present.
-    if origin_detection {
-        if let ConnectionAddress::ProcessLike(Some(creds)) = &peer_addr {
-            for event in event_buffer {
-                if let Some(metric) = event.try_as_metric_mut() {
-                    metric
-                        .metadata_mut()
-                        .origin_entity_mut()
-                        .set_process_id(creds.pid as u32);
-                }
-            }
+fn handle_metric_packet(
+    packet: MetricPacket, event_buffer: &mut EventBuffer, multitenant_strategy: &MultitenantStrategy,
+    origin_detection: bool, peer_addr: &ConnectionAddress, source_metrics: &Metrics,
+) {
+    let metric_metadata = build_metric_metadata_from_packet(&packet, origin_detection, peer_addr);
+    let mut context_resolver = multitenant_strategy.get_context_resolver_for_origin(metric_metadata.origin_entity());
+
+    // Try resolving the context first, since we might need to bail if we can't.
+    let context_ref = context_resolver.create_context_ref(packet.metric_name, &packet.tags);
+    let context = match context_resolver.resolve(context_ref) {
+        Some(context) => context,
+        None => {
+            source_metrics.failed_context_resolve_total().increment(1);
+            // TODO: maybe be louder here, but could be high frequency
```

Comment:
I think it's fine not to log/be louder.

To your point, this could be very high frequency and throwing a bunch of extra stuff at stdout is likely to only slow things down/consume more memory anyways.